### PR TITLE
[highcharts] Allow null y values in datapoints

### PR DESCRIPTION
**Pull request by dyladan**
https://api.highcharts.com/highcharts/series.line.data

Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] Add or edit tests to reflect the change. (Run with `npm test`.)
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present).

If changing an existing definition:
- [x] Provide a URL to documentation or source code which provides context for the suggested changes: https://api.highcharts.com/highcharts/series.line.data
- [ ] Increase the version number in the header if appropriate.
- [ ] If you are making substantial changes, consider adding a `tslint.json` containing `{ "extends": "dtslint/dt.json" }`.

### CODE REVIEW

File: types/highcharts/index.d.ts [hunk 1]
```diff
@@ -5910,7 +5910,7 @@ declare namespace Highcharts {
          *            [3, 7, 8]
          *
          */
-        data?: Array<number | [number, number] | [string, number] | [string, number, number] | [number, number, number] | DataPoint>;
+        data?: Array<number | null | [number, number] | [number, null] | [string, number] | [string, number, number] | [number, number, number] | DataPoint | null>;
```
Comment (dolanmiu) [inline types/highcharts/index.d.ts]: It doesn't say anywhere in the documentation that data can be null. Please revert.

File: types/highcharts/index.d.ts [hunk 2]
```diff
@@ -6134,7 +6134,7 @@ declare namespace Highcharts {
         /**
          * The y value of the point.
          */
-        y?: number;
+        y?: number | null;
```
Comment (dolanmiu) [inline types/highcharts/index.d.ts]: Correct, y can be null:

https://github.com/highcharts/highcharts/blob/8f802d8c7/js/parts/Series.js#L5730-#L5736

**Review by dolanmiu — CHANGES_REQUESTED** [review]
Revert the `data?:` change

`y` change seems ok

File: types/highcharts/index.d.ts (hunk 1 above)
Comment (dyladan) [reply types/highcharts/index.d.ts]: From the documentation:

> An array of numerical values. In this case, the numerical values will be interpreted as `y` options

> An array of arrays with 2 values. In this case, the values correspond to `x,y`.

In addition the type annotation at https://github.com/highcharts/highcharts/blob/8f802d8c7/js/parts/Series.js#L5613 seems to allow it

> `* @type      {Array<number|Array<(number|string),(number|null)>|null|*>}`

Comment (dyladan) [reply types/highcharts/index.d.ts]: Examples:

Data with number and null: https://jsfiddle.net/6vptua08/
Data with [number,number] and [number,null] pairs: https://jsfiddle.net/gzya3e7c/
Data with [string, number] and [string, null] (x value is inferred, in this case 1 per point): https://jsfiddle.net/gzya3e7c/2/

**Comment by dolanmiu** [comment]
My bad, I think you are right

**Review by dolanmiu — APPROVED** [review]

**Comment by dolanmiu** [comment]
Approved

**Pull request merged**